### PR TITLE
Update ponylang/regex dependency to version 1.1.0

### PR DESCRIPTION
**Pull request by adeel41**
So that people on Windows can use the `make.ps1` to build `pcre2-8.lib`

### CODE REVIEW

**Comment by SeanTAllen** [comment]
VERSION is automatically changed when there's a glob release. The change to it should be reverted.

**Comment by ponylang-main** [comment]
Hi @adeel41,

The changelog - changed label was added to this pull request; all PRs with a changelog label need to have release notes included as part of the PR. If you haven't added release notes already, please do.

Release notes are added by creating a uniquely named file in the `.release-notes` directory. We suggest you call the file `19.md` to match the number of this pull request.

The basic format of the release notes (using markdown) should be:

```
## Title

End user description of changes, why it's important,
problems it solves etc.

If a breaking change, make sure to include 1 or more
examples what code would look like prior to this change
and how to update it to work after this change.
```

Thanks.

**Comment by SeanTAllen** [comment]
I don't think that the "can build on windows for a dependency is important" for the release notes @adeel41.

You will still have to build it by hand, yes? There's nothing in this PR that updates the process for this to update the build process. In the end this is only a dependency change.

File: .release-notes/19.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+## Bump up ponylang/regex version to 1.1.0
```
Comment (SeanTAllen) [inline .release-notes/19.md]:
```suggestion
## Update to ponylang/regex version 1.1.0
```

**Comment by SeanTAllen** [comment]
Re: release notes.

I think something like this is better for an end user.

```
## Update ponylang/regex dependency to 1.1.0

This change is a small convenience for Windows users and otherwise provides no change in functionality.

ponylang/regex brings no functionality changes, however, it makes the build process slightly for the dependency easier for Windows users. Note, ponylang/regex still has a dependency on a 3rd party library that still has to be built by hand.
```

**Comment by SeanTAllen** [comment]
Thanks @adeel41!

Once CI passes for this, I'll merge it and cut a new glob release. Everything should be done very shortly.

**Comment by adeel41** [comment]
Copy pasted what you suggested @SeanTAllen exception one small change. Markupdown linter complained last time that the first line should be a top level heading i.e. `#` instead of `##` section heading

**Comment by SeanTAllen** [comment]
The linter is incorrect for this, I'll need to change the linter to ignore that in another PR, please make it ##.

**Comment by SeanTAllen** [comment]
Thanks @adeel41. I just pushed a change so the linter ignores release notes going forward, that was an oversight when it was set up, I found the same issue in at least one other repo and fixed there as well.

**Comment by SeanTAllen** [comment]
Markdown linting will fail on this due to incorrect setup, merging.

**Pull request merged**